tools/config/README.md

```diff
@@ -0,0 +1,44 @@
+# Common scalar build settings.
+
+## Overview
+
+These definitions are only for use in Bazel built-in tools.  In all
+other cases, use flag type definitions from github.com/bazelbuild/bazel-skylib.
+
+<a name="basic-example"></a>
+## Basic Example
+
+### //tools/my_pkg/BUILD
+
+```python
+load("//tools/config:common_settings.bzl", "bool_flag")
+
+bool_flag(
+    name = "experimental_new_behavior",
+    build_setting_default = False,
+    visibility = ["//visibility:public"],
+)
+```
+
+### //tools/my_pkg/myrule.bzl
```

Comment:
Not pressing, but I think big parts of the real power of build settings lies in their ability to be set on the command line (if they can) and their ability to interact with transitions. It would be sweet if this readme reflected that.